pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -124,6 +141,95 @@ func (s secretResource) createSecret(name, label string, identities []byte) erro
 	return nil
 }
 
+func (s secretResource) reconcileKeystoreSecret() (*reconcile.Result, error) {
+	i := s.infinispan
+	if !i.IsEncryptionEnabled() {
+		return nil, nil
+	}
+
+	keystoreSecret := &corev1.Secret{}
+	if result, err := kube.LookupResource(i.GetKeystoreSecretName(), i.Namespace, keystoreSecret, s.client, s.log); result != nil {
+		return result, err
+	}
+
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), s.client, keystoreSecret, func() error {
+		return s.setControllerRef(keystoreSecret)
+	})
+	if err != nil {
+		return &reconcile.Result{}, err
+	}
+	return nil, nil
+}
+
+func (s secretResource) reconcileTruststoreSecret() (*reconcile.Result, error) {
+	i := s.infinispan
+	if !i.IsClientCertEnabled() {
+		return nil, nil
+	}
+
+	trustSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      i.GetTruststoreSecretName(),
+			Namespace: i.Namespace,
+		},
+	}
+
+	if trustSecret.Name == "" {
+		// It's not possible for client cert to be configured if no truststore secret is provided by the user
+		// as no certificates are available to be added to the server's truststore, resulting in no client's
+		// being able to authenticate with the server.
+		return &reconcile.Result{}, fmt.Errorf("Field 'ClientCertSecretName' must be provided for '%s' or '%s' to be configured",
```

Comment:
Tried to run `deploy/cr/minimal/cr_minimal.yaml` and got following error
`Secret resource 'example-infinispan-generated-operator-secret' not ready`
It seems to name was auto-generated, but it's not needed for customer provided fields.

Filed name should start with the lower case.
```suggestion
		return &reconcile.Result{}, fmt.Errorf("Field 'clientCertSecretName' must be provided for '%s' or '%s' to be configured",
```

Reply:
This was my fault for accidentally commiting my changes to `cr_minimal.yaml`. Changes reverted and I have updated the error message.

Reply:
Now error `Secret resource '' not ready`. I think we could validate early and say that if endpointEncryption type is Secret, secret field should be provided.

Reply:
Updated so that an error is thrown in `infinispan_controller.go` if `.Spec.Security.EndpointEncryption.CertSecretName == ""`